src/java.base/share/classes/java/lang/reflect/Constructor.java

```diff
@@ -656,13 +657,22 @@ public AnnotatedType getAnnotatedReceiverType() {
             return null;
         }
 
+        TypeVariable<?>[] v = enclosingClass.getTypeParameters();
```

Comment:
Can you push down most or all of this to resolveOwner? It can then either return a Class instance or an appropriate Type instance.

Reply:
Pushed it down to the factory.